core/state-machine/src/testing.rs

```diff
@@ -48,6 +46,20 @@ pub struct TestExternalities<H: Hasher<Out=H256>=Blake2Hasher, N: ChangesTrieBlo
 }
 
 impl<H: Hasher<Out=H256>, N: ChangesTrieBlockNumber> TestExternalities<H, N> {
+
+	fn ext(&mut self) -> Ext<H, N, InMemory<H>, ChangesTrieInMemoryStorage<H, N>> {
+		Ext {
+			overlay: &mut self.overlay,
+			backend: &self.backend,
+			_phantom: std::marker::PhantomData,
+			storage_transaction: None,
+			changes_trie_storage: Some(&self.changes_trie_storage),
+			changes_trie_transaction: None,
+			id: 0,
+			extensions: None,
+		}
```

Comment:
the only difference with Ext::new is that `id` is not random isn't it ? is `id` important to be zero here ? otherwise just use Ext::new